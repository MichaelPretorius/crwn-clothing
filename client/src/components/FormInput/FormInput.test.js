import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './FormInput';

describe('FormInput', () => {
  let wrapper;
  let mockHandleChange;
  const mockLabel = 'email'

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: mockLabel,
      value: 'test@gmail.com',
      handleChange: mockHandleChange
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  test('should render FormInput component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should call handleChange method when input changes', () => {
    wrapper.find('FormInputContainer').simulate('change');

    expect(mockHandleChange).toBeCalled();
  });

  test('should render FormInputLabel if there is a label', () => {
    expect(wrapper.find('FormInputLabel').text()).toEqual(mockLabel);
  });

  test('should not render FormInputLabel if there is no label', () => {
    const mockNewProps = {
      label: '',
      value: 'test@gmail.com',
      handleChange: mockHandleChange
    };

    const newWrapper = shallow(<FormInput {...mockNewProps} />);

    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });
});