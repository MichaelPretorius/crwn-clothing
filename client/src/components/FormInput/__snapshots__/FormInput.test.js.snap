// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render FormInput component 1`] = `
"<styled.div>
  <FormInputContainer onChange={[Function: mockConstructor]} value=\\"test@gmail.com\\" />
  <FormInputLabel className=\\"shrink\\">
    email
  </FormInputLabel>
</styled.div>"
`;
